### PR TITLE
Update docs to 1.17.0, point to new annotation coordinates

### PR DESCRIPTION
**Pull request by jack-berg**

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Review by chalin — APPROVED** [review]
Thanks all!

File: content/en/docs/instrumentation/java/automatic/annotations.md [hunk 1]
```diff
@@ -12,17 +12,17 @@ custom code without doing too much code change.
...
 
 ```xml
 <dependencies>
   <dependency>
-    <groupId>io.opentelemetry</groupId>
-    <artifactId>opentelemetry-extension-annotations</artifactId>
-    <version>1.16.0</version>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-instrumentation-annotations</artifactId>
+    <version>1.17.0</version>
```
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/annotations.md:25]: it's `1.17.0-alpha`, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/6494

Comment (chalin) [reply content/en/docs/instrumentation/java/automatic/annotations.md:25]: Oops! @jack-berg - can you submit an update?

Comment (jack-berg) [reply content/en/docs/instrumentation/java/automatic/annotations.md:25]: Oops! Fixed in #1651.

**Pull request merged**